R/stroke.R

```diff
@@ -243,43 +259,62 @@ to_linestring <- function(node_id, nodes) {
 }
 
 #' @noRd
-merge_lines  <- function(nodes, segments, links, from_edge = NULL) {
+traverse_segments <- function(start_node, start_link, start_segment,
+                              direction, segments, links, is_segment_used) {
+  node <- start_node
+  link <- start_link
+  segment <- start_segment
+  stroke <- c()
+
+  while (TRUE) {
+    if (is.na(link) || is_segment_used[link]) break
+    node <- get_next_node(node, link, segments)
+    if (direction == "forward") {
+      stroke <- c(node, stroke)
+    } else {
+      stroke <- c(stroke, node)
+    }
```

Comment:
what about  the following: we drop the `direction` argument and simply  append at the top (`stroke <- c(node, stroke)`) for both forward and backward direction. Then we run `rev` on the result of this function for the backward direction? (something like `backward_stroke <- rev(backward_result$stroke)`)